packages/smooth_app/lib/pages/preferences_v2/roots/dev_mode_root.dart

```diff
@@ -0,0 +1,666 @@
+            state:
+                userPreferences.getFlag(userPreferencesFlagProductListImport) ??
+                    false,
+            onToggle: (bool value) async {
+              await userPreferences.setFlag(
+                  userPreferencesFlagProductListImport, value);
+
+              if (!context.mounted) {
+                return;
+              }
+
+              _showSuccessMessage(context, appLocalizations);
+            },
+          ),
+        ],
+      ),
+    ];
+  }
+
+  ScaffoldFeatureController<SnackBar, SnackBarClosedReason> _showSuccessMessage(
+          BuildContext context, AppLocalizations appLocalizations) =>
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          content: Text(appLocalizations.dev_preferences_button_positive),
+        ),
+      );
+
+  Future<void> _changeTestEnvDomain(
+    BuildContext context,
+    UserPreferences userPreferences,
+    AppLocalizations appLocalizations,
+  ) async {
+    _textFieldController.text =
+        userPreferences.getDevModeString(userPreferencesTestEnvDomain) ??
+            uriHelperFoodTest.domain;
+    final bool? result = await showDialog<bool>(
+      context: context,
+      builder: (final BuildContext context) => SmoothAlertDialog(
+        title: appLocalizations.dev_preferences_test_environment_dialog_title,
+        body: TextField(controller: _textFieldController),
+        negativeAction: SmoothActionButton(
+          text: appLocalizations.cancel,
+          onPressed: () => Navigator.pop(context, false),
+        ),
+        positiveAction: SmoothActionButton(
+          text: appLocalizations.okay,
+          onPressed: () => Navigator.pop(context, true),
+        ),
+      ),
+    );
+    if (result == true) {
+      await userPreferences.setDevModeString(
+          userPreferencesTestEnvDomain, _textFieldController.text);
+      ProductQuery.setQueryType(userPreferences);
+    }
+  }
+
+  Future<void> _changeFolksonomyHost(
+    BuildContext context,
+    UserPreferences userPreferences,
+    AppLocalizations appLocalizations,
+  ) async {
+    _textFieldController.text = ProductQuery.uriFolksonomyHelper.host;
+    final String? result = await showDialog<String>(
+      context: context,
+      builder: (final BuildContext context) => SmoothAlertDialog(
+        title: 'Folksonomy host',
+        body: TextField(controller: _textFieldController),
+        negativeAction: SmoothActionButton(
+          text: appLocalizations.cancel,
+          onPressed: () => Navigator.pop(context),
+        ),
+        positiveAction: SmoothActionButton(
+          text: appLocalizations.okay,
+          onPressed: () => Navigator.pop(context, _textFieldController.text),
+        ),
+      ),
+    );
+    if (result != null) {
+      await userPreferences.setDevModeString(
+        userPreferencesFolksonomyHost,
+        result,
+      );
+      ProductQuery.setQueryType(userPreferences);
+    }
+  }
+}
```

Comment:
666 lines? That's a lot.
Of course other things could be said about [this specific number](0https://www.youtube.com/watch?v=WxnN05vOuSM), but that's another story...